File: codehost/github/converters/file.go
package converters

import (
	"github.com/google/go-github/v52/github"
	"github.com/reviewpal/reviewpal/codehost/target"
)

func CommitFilesFromGithubCommitFiles(fs []*github.CommitFile) []target.CommitFile {
	files := make([]target.CommitFile, len(fs))
	for i, f := range fs {
		files[i] = CommitFileFromGithubCommitFile(f)
	}

	return files
}

func CommitFileFromGithubCommitFile(f *github.CommitFile) target.CommitFile {
	return target.CommitFile{
		Filename: f.GetFilename(),
		Patch:    f.GetPatch(),
	}
}

File: src/codehost/github/converters/file.rs
use octocrab::models::repos::DiffEntry as GithubCommitFile;
use crate::codehost::target::file::CommitFile;

pub fn commit_file_from_github_commit_file(file: GithubCommitFile) -> CommitFile {
        CommitFile {
            filename: file.filename.clone(),
            patch: file.patch.clone().unwrap_or("".to_string()),
        }
}
